### PR TITLE
docs: Document user overrides (part 2)

### PR DESCRIPTION
**Pull request by sbernauer**
Fixup of https://github.com/stackabletech/documentation/pull/415

### CODE REVIEW

File: modules/concepts/pages/overrides.adoc [hunk 1]
```diff
@@ -37,7 +43,7 @@ The properties will be formatted and escaped correctly into the file format used
 
 == Environment variable overrides
 
-For a role or role group, at the same level of `config`, you can specify `envOverrides` for any env variable
+For a xref:roles-and-role-groups.adoc[role or role group], at the same level of `config`, you can specify `envOverrides` for any env variable
 
 An example for an HDFS clusters looks as follows:
```
Comment (maltesander) [inline modules/concepts/pages/overrides.adoc:48]:
```suggestion
An example for an HDFS cluster looks as follows:
```

File: modules/concepts/pages/overrides.adoc [hunk 2]
```diff
@@ -63,12 +69,12 @@ spec:
...
-The environment variables you can set depend on the specific product.
+You can any environment variables, but every specific product does support a different set of environment variables.
 All override property values must be strings.
 
 == Pod overrides
 
-For a role or role group, at the same level of `config`, you can specify `podOverrides` for any of the attributes you can configure on a Pod.
+For a xref:roles-and-role-groups.adoc[role or role group], at the same level of `config`, you can specify `podOverrides` for any of the attributes you can configure on a Pod.
 
 An example for an HDFS clusters looks as follows:
```
Comment (maltesander) [inline modules/concepts/pages/overrides.adoc:79]:
```suggestion
An example for an HDFS cluster looks as follows:
```

File: modules/concepts/pages/overrides.adoc [hunk 3]
```diff
@@ -108,3 +114,12 @@ The priority of how to construct the final Pod submitted to Kubernetes looks as
 
 Each of these are combined top to bottom using a deep merge.
 The exact merge algorithm is described in the https://arnavion.github.io/k8s-openapi/v0.18.x/k8s_openapi/trait.DeepMerge.html[k8s-openapi docs], which basically tries to mimic the way Kubernetes merges patches onto objects.
+
+The `podOverrides` will be merged onto the following resources the operators deploy:
+
+* StatefulSets containing the products (most of the products)
+* DaemonSets containing the products (currently only OPA)
```
Comment (maltesander) [inline modules/concepts/pages/overrides.adoc:121]: Just for completeness i would mention deployments as well (even if there is no usecase currently, we already have that trino deployment discussion somewhere?)

File: modules/concepts/pages/overrides.adoc [hunk 4]
```diff
@@ -63,12 +69,12 @@ spec:
         replicas: 1
 ----
 
-The environment variables you can set depend on the specific product.
+You can any environment variables, but every specific product does support a different set of environment variables.
```
Comment (maltesander) [inline modules/concepts/pages/overrides.adoc:72]:
```suggestion
You can set any environment variable, but every specific product does support a different set of environment variables.
```

File: modules/concepts/pages/overrides.adoc [hunk 5]
```diff
@@ -1,12 +1,18 @@
 = Overrides
 
+The Stackable Operators configure the products they are operating with sensible defaults, and required settings to enable connectivity and security when enabled.
```
Comment (maltesander) [inline modules/concepts/pages/overrides.adoc:3]:
```suggestion
The Stackable Operators configure the products they are operating with sensible defaults and required settings to enable connectivity and security.
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Minor things. LGTM otherwise!

File: modules/concepts/pages/overrides.adoc (hunk 3 above)
Comment (sbernauer) [reply modules/concepts/pages/overrides.adoc:121]: Please have a look

File: modules/concepts/pages/overrides.adoc [hunk 6]
```diff
@@ -1,12 +1,18 @@
...
 
-The cluster definitions also supports overriding configuration aspects, either per role or per role group, where the more specific override (role group) has precedence over the less specific one (role).
+The cluster definitions also supports overriding configuration aspects, either per xref:roles-and-role-groups.adoc[role or per role group], where the more specific override (role group) has precedence over the less specific one (role).
 
 == Config overrides
 
-For a role or role group, at the same level of `config`, you can specify `configOverrides` for any of the configuration files the product uses.
+For a xref:roles-and-role-groups.adoc[role or role group], at the same level of `config`, you can specify `configOverrides` for any of the configuration files the product uses.
 
 An example for an HDFS clusters looks as follows:
```
Comment (maltesander) [inline modules/concepts/pages/overrides.adoc:17]: Sorry spotted another old one... 
```suggestion
An example for an HDFS cluster looks as follows:
```

**Review by maltesander — APPROVED** [review]
LGTM!

**Comment by sbernauer** [comment]
Many thanks!

**Review by maltesander — APPROVED** [review]

**Pull request merged**